### ISSUE TITLE
Integrate with Photonics 3D in a way as yet to be determined.

### ISSUE THREAD

**Issue by DanielJoyce** [Microtome/microtome#11]
Sounds cool, all for it. I also have Polymer UI experience, and I have to say Polymer + Typescript makes UI development pretty fast and easy. Data binding, code error checking, and three.js is a lot simpler to use with typing information

If we do integrate, the slicer will stay TypeScript (loosey-goosey typing in JS is madness!) though the resulting Javascript artifacts should work just fine no matter how you want to use them.

So what I could do is split out a library containing just the current slicer code and other dependencies.

**Comment by WesGilster** [comment]
I completely agree. I'm generally pretty skeptical of developers who say that JS is fun or is somehow a good language. It's certainly a very important language though.

**Comment by DanielJoyce** [comment]
Alright started back on this. Trying to get the new slicer out into a library and cleaning up the build. Gulp is a shit pile.

**Comment by WesGilster** [comment]
Great to hear! If you install Photonic 3d it's got quite a nice restful interface, with detailed control of every printer function you might need. If something is missing or broken, feel free to open a bug or feature request. You'll find we are quite responsive. The API is also documented with a swaggerized "try-it" interface by going here:
http://localhost:9091/apidoc/

Which will give you an overview of the 7 main APIs. We also have an undocumented websocket interface for push notifications as well.
![image](https://cloud.githubusercontent.com/assets/2424628/20391528/64fe57b2-ac99-11e6-838d-60ced83f33e8.png)

You'll probably be interested in the Printer and Printables APIs:
![image](https://cloud.githubusercontent.com/assets/2424628/20391680/0f0488e4-ac9a-11e6-95cb-544e4c1d2570.png)

And the try it interface is like this:
http://localhost:9091/apidoc/#!/printers/deletePrinter
![image](https://cloud.githubusercontent.com/assets/2424628/20391820/a67014aa-ac9a-11e6-84c4-b90c52163ab8.png)

**Comment by DanielJoyce** [comment]
How do I build and run locally? Will need to add a custom page

The main idea is run the slicer in the browser, then ship the images to photonics3d to use as a job

**Comment by WesGilster** [comment]
If you are just interested in getting up and running quickly we have several options here:
https://wiki.photonic3d.com/doku.php?id=installing

If you are looking for a simple download we have our latest stable build here:
https://github.com/area515/Photonic3D/releases

If you are looking for the latest development build you can go to my repo:
https://github.com/WesGilster/Photonic3D/releases

If you are actually wanting to build from source and you are looking for the actual location of the gradle builds for building locally:
https://github.com/area515/Photonic3D/tree/master/host

if you are interested in latching your fork(if you are going to fork) into a continuous integration system you can do that with your repo with this:
https://github.com/area515/Photonic3D/blob/master/.travis.yml

You might also notice that slacer was already added as a git sub module for easy integration as well. I haven't written any of these scripts, so if you have further questions, feel free to ask them on our area515/Photonic3D repo issues.

**Comment by DanielJoyce** [comment]
Okay, so after having yet another kid, and a job changes, and some time to work on the slicer on the train I've gotten things massively cleaned up and simplified.

**Comment by DanielJoyce** [comment]
As opposed to git submodules which are their own flavor of evil, I'd use yarn or npm to pull them in.

**Comment by WesGilster** [comment]
Cool nice job. When are you going to have supports available!? :)
